mainapp/serializers.py

```diff
@@ -0,0 +1,54 @@
+from datetime import timedelta
+
+from django.contrib.auth.password_validation import validate_password
+from django.utils import timezone
+from rest_framework import serializers
+from rest_framework.validators import UniqueValidator
+
+from .models import Profile, User
+
+
+class RegisterSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(
+        required=True, validators=[UniqueValidator(queryset=User.objects.all())]
+    )
+
+    password = serializers.CharField(
+        write_only=True, required=True, validators=[validate_password]
+    )
+    password2 = serializers.CharField(write_only=True, required=True)
+
+    class Meta:
+        model = User
+        fields = ("email", "password", "password2")
+
+    def validate(self, attrs):
+        if attrs["password"] != attrs["password2"]:
+            raise serializers.ValidationError(
+                {"password": "Password fields didn't match."}
+            )
+
+        return attrs
+
+    def create(self, validated_data):
+        user = User.objects.create(
+            email=validated_data["email"],
+        )
+        user.set_password(validated_data["password"])
+        user.save()
+
+        return user
+
+
+class ProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Profile
+        fields = "__all__"
+
+    def validate(self, data):
+        tenminearly = timezone.now() - timedelta(minutes=10)
```

Comment:
ten_min_early. (make variables snake case.) Read about Snake Case and Camel Case